packages/vscode-extension/src/debugging/DebugAdapter.ts

```diff
@@ -150,7 +154,22 @@ export class DebugAdapter extends DebugSession {
             const decodedData = Buffer.from(base64Data, "base64").toString("utf-8");
             const sourceMap = JSON.parse(decodedData);
             const consumer = await new SourceMapConsumer(sourceMap);
-            this.sourceMaps.push([message.params.url, message.params.scriptId, consumer]);
+
+            // This line is here because of a problem with sourcemaps for expo projects,
+            // that was addressed in this PR https://github.com/expo/expo/pull/29463,
+            // unfortunately it still requires changes to metro that were attempted here
+            // https://github.com/facebook/metro/pull/1284 we should monitor the situation
+            // in upcoming versions and if the changes are still not added bump the version below.
+            // more over offset should only be applied to the main bundle sourcemap as other files
+            // (generated during reload) do not contain the prelude causing the issue
+            const shouldApplyOffset =
```

Comment:
This check isn't accurate I think, we don't want to use the offset for the first source map only but for any source map that covers the prelude bit. Meaning that all JS reloads would also have this issue, while we may not handle them correctly. Have you tried using reloadJS with your changes now? Note that when we calculate generated position, we do a full loop over all the maps and if the entry is found in multiple maps, we take the last record. So the newer source map will be considered in that case.